### PR TITLE
Revert the revert of PR #8766

### PR DESCRIPTION
**Pull request by agocke**
This is the same PR as #8766, but it downgrades the packages used by portable assemblies back to the RTM versions so the facades don't change assembly version.

This change has been validated by https://github.com/dotnet/roslyn-internal/pull/594, and then rebased.

/cc @jaredpar @davkean @dotnet/roslyn-compiler @dotnet/roslyn-infrastructure

### CODE REVIEW

**Comment by agocke** [comment]
Woops, looks like I accidentally removed something necessary in the rebase. Will add back.

**Comment by agocke** [comment]
This should be all fixed up now -- missing commits restored and nuget packages updated.

**Comment by davkean** [comment]
Are the eta failures expected?

Can you please open Roslyn.sln and ensure that you haven't introduced any errors/warnings in the error list. The previous version of this added a significant number of errors.

**Comment by jaredpar** [comment]
@davkean ETA failures should be expected.  The ETA job is passing on the internal run.  

I verified on my machine there were no errors in the list using Roslyn.sln.  Did a full build locally.  Ddi see the following warning scroll by, not sure if new

> C:\Program Files (x86)\MSBuild\14.0\bin\Microsoft.Common.CurrentVersion.targets(1819,5): warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.

**Comment by davkean** [comment]
Can you see if the warnings are coming from the same projects here: https://github.com/dotnet/roslyn/issues/8631?

If so, then this didn't introduce them.

**Comment by davkean** [comment]
Actually, they might be new, can you tell what assembly it's complaining about?

File: BuildAndTest.proj [hunk 1]
```diff
@@ -75,13 +76,17 @@
 
       <!-- MakeConst unit tests tracked by https://github.com/dotnet/roslyn/issues/5918 -->
       <TestAssemblies Condition="'$(Test64)' != 'true'" 
-                      Include="$(OutputDirectory)\**\$(IncludePattern)" />
+                      Include="$(OutputDirectory)\$(IncludePattern)" />
```
Comment (davkean) [inline BuildAndTest.proj]: This will stop running project system tests as they don't exist in the root. To avoid regressing this, manually include them (they build to bin\ProjectSystem\Tests), and then we'll change the output later when this in in future to fix this.

Comment (davkean) [inline BuildAndTest.proj]: Actually, maybe TestAssemblies should remove the result of "CoreTestAssemblies", that way we don't ever miss any tests.

Comment (davkean) [inline BuildAndTest.proj]: @tmat and @amcasey Why are we excluding interactive here?

Comment (agocke) [inline BuildAndTest.proj]: @davkean The idea behind the DeployDesktopTestRuntime project is that it isolates a specific set of tests and test dependencies to a particular directory. If other tests are deploying to a different directory I would consider that a bug.

Comment (davkean) [inline BuildAndTest.proj]: Regardless of whether you consider this a bug or not, the project system tests are deploying to a different directory. 
This change appears to cause these tests to stop running, please fix that.

Comment (agocke) [inline BuildAndTest.proj]: @davkean Right, I'm arguing I should fix that by changing the output directory of the project system tests :)

Where does that code live?

Comment (davkean) [inline BuildAndTest.proj]: It's in future, under https://github.com/dotnet/roslyn/tree/future/src/VisualStudio/ProjectSystem. 

You'll need to change to change DeployTestDependencies and the test projects (currently deploy to bin\ProjectSystem\Tests). The product itself deploys into a different folder (bin\ProjectSystem) - so don't change that as you'll cause a build race.

It might be easier to do what I suggested above, and then I'll/you'll fixed it future. I just want to avoid accidentally merging this into future and then tests stop running.

Comment (davkean) [inline BuildAndTest.proj]: Given that you now don't run tests outside of the root or in the core clr folder, can we break the build if there are tests found outside of it?

Comment (davkean) [inline BuildAndTest.proj]: Actually that might be the best way - then we'll never merge this into future without a build break.

Comment (davkean) [inline BuildAndTest.proj]: Do we have bugs? Work planned to update both the signed build and internal builds to also do something similar?

Comment (agocke) [inline BuildAndTest.proj]: @davkean That's a good idea and question about the bug. I may try to do that as a separate PR I merge before this one, though.

**Comment by agocke** [comment]
I've also downgraded some more references to get rid of the warnings.

**Comment by jaredpar** [comment]
:+1:

File: BuildAndTest.proj (hunk 1 above)
Comment (amcasey) [inline BuildAndTest.proj]: @davkean, I'm not sure why they're excluded from 64-bit.  Try unskipping them?

**Comment by agocke** [comment]
test prtest/win/dbg/unit64 please

**Comment by agocke** [comment]
@davkean I've added a commit to fail the build if tests are found outside the specified test directories.

**Comment by agocke** [comment]
test prtest/win/dbg/unit64 please

**Comment by agocke** [comment]
@davkean Because of the exclusion of the interactive tests -- they cause the 64 bit build to fail because they're present in the root, but not included in any exclude list.

**Comment by agocke** [comment]
@davkean How is x86-64 failing? Reading the project.lock.jsons it doesn't look like anything should be referencing System.AppContext 4.0.1 in the Desktop test directory, so 4.0.0 should be the only assembly in there.

**Comment by davkean** [comment]
Hmm. First of all, when you build this locally do you see AppContext 4.01?

**Comment by agocke** [comment]
@davkean Tried it locally -- I get System.AppContext 4.0.0

**Comment by davkean** [comment]
If you look at the error messages - this is coming from CsiCore:
`d:\j\workspace\roslyn_prtest---45fcf4bc\src\Interactive\CsiCore\`. Looks like these are running on desktop, and hence will need binding redirects from 4.0.0 -> 4.0.1 for System.AppContext as it doesn't exist in the framework.

**Comment by davkean** [comment]
Or alternatively, do you need to update AppContext for the CoreCLR binaries?

**Comment by agocke** [comment]
test  prtest/win/dbg/unit64  please

**Comment by agocke** [comment]
@davkean This looks to be all updated and working now. I'll resolve the merge conflicts and re-run tests.

**Comment by davkean** [comment]
:+1: Assuming you re-run Restore.cmd to pick up the new NuGet.exe's project.lock.json.

Thanks for adding the error!

**Comment by jaredpar** [comment]
image
